### ISSUE TITLE
OSGi ERROR osgi.wiring.package=sun.misc

### ISSUE THREAD

**Issue by gadieichhorn** [OpenHFT/Chronicle-Wire#17]
ERROR when loading in wire bundle

Can sun.misc be declared as optional?
What are the exact dependencies I have to load to make this work?

 
`ERROR: Bundle net.openhft.chronicle-wire [80] Error starting file:bundles/net.openhft.chronicle-wire_1.3.4.jar (org.osgi.framework.BundleException: Unresolved constraint in bundle net.openhft.chronicle-wire [80]: Unable to resolve 80.0: missing requirement [80.0] osgi.wiring.package; (&(osgi.wiring.package=net.openhft.chronicle.bytes)(version>=1.2.0)(!(version>=2.0.0))) [caused by: Unable to resolve 81.0: missing requirement [81.0] osgi.wiring.package; (&(osgi.wiring.package=net.openhft.chronicle.core)(version>=1.3.0)(!(version>=2.0.0))) [caused by: Unable to resolve 82.0: missing requirement [82.0] osgi.wiring.package; (osgi.wiring.package=sun.misc)]])`

**Comment by peter-lawrey** [comment]
sun.misc is required, you need to set your OSGI to expose the sun.misc.*
from your JVM.

We have OSGI bundles enabled but don't have tests for it.

**Comment by gadieichhorn** [comment]
This can be an issue with specific implementation of JVM.
http://www.oracle.com/technetwork/java/faq-sun-packages-142232.html 

and now we getting a new errors with range of missing dependencies and at the end there are missing OSGi import/export from bundles

```
`ERROR: Bundle net.openhft.chronicle-bytes [81] Error starting file:bundles/net.openhft.chronicle-bytes_1.2.3.jar (org.osgi.framework.BundleException: Unresolved constraint in bundle net.openhft.chronicle-bytes [81]: Unable to resolve 81.0: missing requirement [81.0] osgi.wiring.package; (&(osgi.wiring.package=net.openhft.chronicle.core)(version>=1.3.0)(!(version>=2.0.0))) [caused by: Unable to resolve 82.0: missing requirement [82.0] osgi.wiring.package; (osgi.wiring.package=sun.nio.ch)])`

ERROR: Bundle net.openhft.chronicle-core [82] Error starting file:bundles/net.openhft.chronicle-core_1.3.0.jar (org.osgi.framework.BundleException: Unresolved constraint in bundle net.openhft.chronicle-core [82]: Unable to resolve 82.0: missing requirement [82.0] osgi.wiring.package; (osgi.wiring.package=sun.reflect))

ERROR: Bundle net.openhft.chronicle-bytes [81] Error starting file:bundles/net.openhft.chronicle-bytes_1.2.3.jar (org.osgi.framework.BundleException: Unresolved constraint in bundle net.openhft.chronicle-bytes [81]: Unable to resolve 81.0: missing requirement [81.0] osgi.wiring.package; (&(osgi.wiring.package=org.xerial.snappy)(version>=1.1.0)(!(version>=2.0.0))))

ERROR: Bundle net.openhft.chronicle-wire [80] Error starting file:bundles/net.openhft.chronicle-wire_1.3.5.jar (org.osgi.framework.BundleException: Unresolved constraint in bundle net.openhft.chronicle-wire [80]: Unable to resolve 80.0: missing requirement [80.0] osgi.wiring.package; (osgi.wiring.package=net.openhft.chronicle.bytes.ref))

org.osgi.framework.BundleException: Unresolved constraint in bundle net.openhft.chronicle-wire [80]: Unable to resolve 80.0: missing requirement [80.0] osgi.wiring.package; (osgi.wiring.package=net.openhft.chronicle.bytes.ref)

```

**Comment by peter-lawrey** [comment]
The osgi bundle configuration needs to be checked. I am about to take off
to Heathrow.  Can reply when I land.

**Comment by lburgazzoli** [comment]
@gadieichhorn what OSGi framework are you on ?

**Comment by lburgazzoli** [comment]
I've committed a number of fixed for OSGi meta-data and not all the bundles start, here the list of bundles and dependencies running in Apache Karaf:

| ID | Status|StartLevel|Version| SymbolicName|
|----|-----------|------|--------------------|-----------------------------------------------------|
| 95 | Active   |  80 | 4.2.1                | jna|
| 96 | Active   |  80 | 4.2.1                | jna-platform|
| 97 | Active   |  80 | 0                    | wrap_mvn_com.squareup_javapoet_1.5.1|
|100 | Active   |  80 | 3.0.2.SNAPSHOT       | OpenHFT :: affinity|
|101 | Active   |  80 | 2.2.2.SNAPSHOT       | OpenHFT :: compiler|
|102 | Active   |  80 | 1.3.1.SNAPSHOT       | OpenHFT :: chronicle-core|
|103 | Active   |  80 | 1.2.4.SNAPSHOT       | OpenHFT :: chronicle-bytes|
|104 | Active   |  80 | 1.3.1.SNAPSHOT       | OpenHFT :: chronicle-threads|
|105 | Active   |  80 | 1.0.4.alpha-SNAPSHOT | OpenHFT :: chronicle-values|
|106 | Active   |  80 | 1.3.5.SNAPSHOT       | OpenHFT :: chronicle-wire|
|107 | Active   |  80 | 1.0.65.SNAPSHOT      | OpenHFT :: chronicle-network|

**Comment by gadieichhorn** [comment]
We use Felix.

**Comment by peter-lawrey** [comment]
Closing for now, assuming it is fixed.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.20.101](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.20.101), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
